packages/codemod/src/transforms/react-default-props-migration.ts

```diff
@@ -0,0 +1,426 @@
+        return <div>{prop1}</div>;
+      }
+    ```
+  */
+  scope
+    .find(j.MemberExpression, {
+      object: { type: 'Identifier', name: 'props' },
+    })
+    .forEach((memberPath: ASTPath<MemberExpression>) => {
+      const property = memberPath.node.property;
+      // Add type guard for Identifier
+      if (
+        property.type === 'Identifier' &&
+        defaultPropsKeys.includes(property.name)
+      ) {
+        j(memberPath).replaceWith(j.identifier(property.name));
+      }
+    });
+
+  /*
+    Next code block replaces props usage in the component body where
+    props is passed as an argument to a function.
+    ```
+      // BEFORE
+      const MyComponent = (props) => {
+        return <div>{getStyles(props)}</div>;
+      }
+
+      // AFTER
+      const MyComponent = ({ prop1, ...props }) => {
+        return <div>{getStyles({ prop1, ...props })}</div>;
+      }
+    ```
+  */
+  scope
+    .find(j.CallExpression, {
+      arguments: [{ type: 'Identifier', name: 'props' }],
+    })
+    .forEach((callPath: ASTPath<CallExpression>) => {
+      // Create a destructured object
+      const properties = [
+        ...defaultPropsKeys.map((key) => {
+          const id = j.identifier(key);
+          const newProp = j.property('init', id, id);
+          newProp.shorthand = true;
+          return newProp;
+        }),
+        j.spreadElement(j.identifier('props')),
+      ];
+
+      const objectExpression = j.objectExpression(properties);
+
+      // Replace the 'props' argument with the destructured object
+      callPath.node.arguments[0] = objectExpression;
+    });
+}
+
+/*
+  We need to make sure the component type definition is updated to reflect the
+  props that are now optional.
+  Example:
+  ```
+  // BEFORE
+  type MyComponentProps = {
+    prop1: string;
+    prop2: string;
+    prop3: string;
+  }
+  function MyComponent(props: MyComponentProps) { ... }
+  MyComponent.defaultProps = {
+    prop1: 'default value',
+  }
+
+  // AFTER
+  type MyComponentProps = {
+    prop1?: string;
+    prop2: string;
+    prop3: string;
+  }
+  function MyComponent({ prop1, ...props }: MyComponentProps) { ... }
+  ```
+*/
+function updateComponentTypes({
+  j,
+  root,
+  typeName,
+  destructuredKeys,
+}: {
+  j: JSCodeshift;
+  root: Collection;
+  typeName: string;
+  destructuredKeys: string[];
+}) {
+  // Find the type definition of the component props
+  root
+    .find(j.TSTypeAliasDeclaration)
+    .forEach((typePath: ASTPath<TSTypeAliasDeclaration>) => {
+      if (typePath.node.id.name === typeName) {
+        const typeAnnotation = typePath.node.typeAnnotation;
+
+        if (typeAnnotation.type === 'TSTypeLiteral') {
+          typeAnnotation.members.forEach((member) => {
+            if (
+              member.type === 'TSPropertySignature' &&
+              member.key.type === 'Identifier' &&
+              destructuredKeys.includes(member.key.name)
+            ) {
+              member.optional = true;
```

Comment:
This is where we make the type member of the prop optional when it is part of the prop right ?

Reply:
Yes. I need to add another comment.